portalnet/src/metrics/overlay.rs

```diff
@@ -1,81 +1,84 @@
+use crate::types::messages::{Request, Response};
 
-/// General Metrics Strategy (wip)
-/// - Each module should maintain its own metrics reporter
-/// - When possible, use the lazy_static! approach
-/// - - https://romankudryashov.com/blog/2021/11/monitoring-rust-web-application/
-/// - - https://github.com/sigp/lighthouse/blob/c3a793fd73a3b11b130b82032904d39c952869e4/beacon_node/lighthouse_network/src/metrics.rs
-
-/// Overlay Service Metrics Reporter
-#[derive(Clone, Debug)]
+/// Contains metrics reporters for use in the overlay network
+/// (eg. `portalnet/src/overlay.rs` & `portalnet/src/overlay_service.rs`).
+/// Metric types reported here include protocol messages, utp transfers,
+/// and content validation.
+#[derive(Clone)]
 pub struct OverlayMetrics {
-    protocol: ProtocolLabel,
-    message_count: IntCounterVec,
-    utp_outcome_count: IntCounterVec,
-    utp_active_count: IntGaugeVec,
-    validation_count: IntCounterVec,
+    pub message_count: IntCounterVec,
+    pub utp_outcome_count: IntCounterVec,
+    pub utp_active_gauge: IntGaugeVec,
+    pub validation_count: IntCounterVec,
 }
 
 impl OverlayMetrics {
-    pub fn new(protocol: &ProtocolId) -> Self {
-        let message_count_options = opts!(
-            "trin_message_total",
-            "count all network messages sent and received"
-        );
-        let message_count_labels = &["protocol", "direction", "type"];
-        let message_count =
-            OverlayMetrics::register_counter_metric(message_count_options, message_count_labels);
-
-        let utp_outcome_count_options = opts!(
-            "trin_utp_outcome",
-            "track success rate for all utp transfers outbound and inbound"
-        );
-        let utp_outcome_count_labels = &["protocol", "direction", "outcome"];
-        let utp_outcome_count = OverlayMetrics::register_counter_metric(
-            utp_outcome_count_options,
-            utp_outcome_count_labels,
-        );
-
-        let utp_active_count_options = opts!(
-            "trin_utp_active",
-            "count all active utp transfers outbound and inbound"
-        );
-        let utp_active_count_labels = &["protocol", "direction"];
-        let utp_active_count = OverlayMetrics::register_gauge_metric(
-            utp_active_count_options,
-            utp_active_count_labels,
-        );
-
-        let validation_count_options = opts!(
-            "trin_validation_total",
-            "count all content validations successful and failed"
-        );
-        let validation_count_labels = &["protocol", "success"];
-        let validation_count = OverlayMetrics::register_counter_metric(
-            validation_count_options,
-            validation_count_labels,
-        );
-
-        Self {
-            protocol: protocol.into(),
+    pub fn new(registry: &Registry) -> anyhow::Result<Self> {
+        let message_count = register_int_counter_vec_with_registry!(
+            opts!(
+                "trin_message_count",
```

Comment:
`total` is the preferred suffix for a generic aggregating counter, according to this Prometheus naming "Best Practices" page: https://prometheus.io/docs/practices/naming/